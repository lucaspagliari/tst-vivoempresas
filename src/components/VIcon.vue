<script setup lang="ts">
import { computed } from "vue";
import { defineAsyncComponent } from "vue";

const props = defineProps<{
  icon: string;
  color?: string;
}>();

const currentIcon = computed(() =>
  defineAsyncComponent(
    () => import(`../assets/svg/${props.icon}.svg?component`)
  )
);
</script>
<template>
  <span class="v-icon">
    <component :is="currentIcon"  />
  </span>
</template>

<style scoped>
.v-icon {
  stroke: v-bind(color);
}
</style>
