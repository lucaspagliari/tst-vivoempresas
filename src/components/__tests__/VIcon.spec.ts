import VIcon from "../VIcon.vue";
import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/vue";

describe("Component VIcon", () => {
  it.each([
    { icon: "menu" },
    { icon: "chip" },
    { icon: "close" },
    { icon: "sim_card" },
  ])("should receive $icon as a prop and match spnapshot", ({ icon }) => {
    const c = render(VIcon, {
      props: { icon },
    });

    expect(c.html()).toMatchSnapshot();
  });
});
