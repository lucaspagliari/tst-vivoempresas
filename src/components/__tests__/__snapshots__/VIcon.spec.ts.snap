// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component VIcon > should receive 'chip' as a prop and match spnapshot 1`] = `
"<transition-stub data-v-5f2d8097="" name="fade" mode="out-in" appear="false" persisted="false" css="true">
  <!---->
</transition-stub>"
`;

exports[`Component VIcon > should receive 'close' as a prop and match spnapshot 1`] = `
"<transition-stub data-v-5f2d8097="" name="fade" mode="out-in" appear="false" persisted="false" css="true">
  <!---->
</transition-stub>"
`;

exports[`Component VIcon > should receive 'menu' as a prop and match spnapshot 1`] = `
"<transition-stub data-v-5f2d8097="" name="fade" mode="out-in" appear="false" persisted="false" css="true">
  <!---->
</transition-stub>"
`;

exports[`Component VIcon > should receive 'sim_card' as a prop and match spnapshot 1`] = `
"<transition-stub data-v-5f2d8097="" name="fade" mode="out-in" appear="false" persisted="false" css="true">
  <!---->
</transition-stub>"
`;
